src/Store.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useReducer } from 'react';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+
+type Action =
+  | { type: 'ADD_TODO'; payload: string }
+  | { type: 'UPDATE_TODO'; payload: { id: number; title: string } }
+  | { type: 'REMOVE_TODO'; payload: number }
+  | { type: 'TOGGLE_TODO'; payload: number }
+  | { type: 'TOGGLE_ALL'; payload: boolean }
+  | { type: 'CLEAR_COMPLETED' }
+  | { type: 'SET_FILTER'; payload: Status }
+  | { type: 'SET_TODOS'; payload: Todo[] };
+
+type State = {
+  todos: Todo[];
+  filter: Status;
+};
+
+const initialState: State = {
+  todos: [],
+  filter: Status.All,
+};
+
+export const TodosContext = React.createContext<{
+  state: State;
+  dispatch: React.Dispatch<Action>;
+}>({
+  state: initialState,
+  dispatch: () => {},
+});
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case 'ADD_TODO':
```

Comment:
I think it would be a good idea to create enum of actionType names and use it everywhere